File: 922.按奇偶排序数组-ii.py
#
# @lc app=leetcode.cn id=922 lang=python3
#
# [922] 按奇偶排序数组 II
#

# @lc code=start
class Solution:
    def sortArrayByParityII(self, A: List[int]) -> List[int]:
        j = 1
        for i in range(0, len(A), 2):
            if A[i] %2 :
                while A[j] %2:
                    j += 2
                A[i], A[j] = A[j], A[i]
        return A
        
# @lc code=end

File: 922.按奇偶排序数组-ii.cpp
/*
 * @lc app=leetcode.cn id=922 lang=cpp
 *
 * [922] 按奇偶排序数组 II
 */

// @lc code=start
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& A) {
        int left = 0, right = 1;
        while(left < A.size() and right < A.size()){
            while(left < A.size() and A[left] % 2 ==0){
                left+=2;
            }
            while(right <A.size()  and A[right] % 2 == 1){
                right += 2;
            }
            if(left >= A.size() or right >= A.size())
                break;
            swap(A[left], A[right]);
            left += 2;
            right += 2;
        }
        return A;
    }
};
// @lc code=end

